packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -1712,7 +1754,7 @@ class RequestFocusIntent extends Intent {
 class RequestFocusAction extends Action<RequestFocusIntent> {
   @override
   void invoke(RequestFocusIntent intent) {
```

Comment:
The `RequestFocusIntent` could take an optional `TraversalRequestFocusCallback` in its constructor, which you could call here.  That would let the `RequestFocusAction` be customizable in the same way.

Of course, It would also need a test.  Feel free to leave that for another PR if you don't have time.

Reply:
That's a good catch, I'll do it on this PR since it makes the most sense.